packages/beacon-node/src/chain/regen/queued.ts

```diff
@@ -71,6 +71,40 @@ export class QueuedStateRegenerator implements IStateRegenerator {
     return this.stateCache.get(stateRoot);
   }
 
+  getPreStateSync(block: allForks.BeaconBlock): CachedBeaconStateAllForks | null {
```

Comment:
Please take a closer look at `getPreStateSync()` and its usage. Pretty much moved the majority of `getPreState()` to this function @tuyennhv @wemeetagain @nflaig

Reply:
yes looks good to me